packages/relayer/processor/process_message.go

```diff
@@ -1,43 +1,86 @@
-package message
+package processor
 
 import (
 	"context"
 	"encoding/hex"
+	"encoding/json"
 	"fmt"
 	"log/slog"
 	"math/big"
 	"strings"
 	"time"
 
+	"github.com/cenkalti/backoff"
 	"github.com/ethereum/go-ethereum/accounts/abi"
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/pkg/errors"
 	"github.com/taikoxyz/taiko-mono/packages/relayer"
-	"github.com/taikoxyz/taiko-mono/packages/relayer/contracts/bridge"
+	"github.com/taikoxyz/taiko-mono/packages/relayer/bindings/bridge"
+	"github.com/taikoxyz/taiko-mono/packages/relayer/queue"
 )
 
+var (
+	errUnprocessable = errors.New("message is unprocessable")
+)
+
+func (p *Processor) eventStatusFromMsgHash(
+	ctx context.Context,
+	gasLimit *big.Int,
+	signal [32]byte,
+) (relayer.EventStatus, error) {
+	var eventStatus relayer.EventStatus
+
+	messageStatus, err := p.destBridge.GetMessageStatus(nil, signal)
```

Comment:
will suggest adding a timeout for the `ctx` and making `&bind.CallOpts{Context: ctx}` as the first parameter.

Reply:
done, added a new flag for ETHClientTimeout